### DISCUSSION TITLE
Click again to deselect an item in List

### DISCUSSION THREAD

**Discussion by aneeskA** [fyne-io/fyne#4962] [General] [answered]
Is there a way to make the current selected item in the List to be deselected on the next click? The `OnSelected` callback is not getting called on the subsequent clicks.

**Comment by andydotxyz** [comment]
That's not really how the selection model works, perhaps you could expand on what you are trying to do?

If you want togglable items you could use Check widget?

**Reply by aneeskA** [reply]
Let me explain what I am looking for. I am developing a desktop app that shows a list of topics. When the user selects a topic, I changes to the rest of the UI. I also want to undo those actions when they click on it again. I was hoping to catch the second one with the OnSelected callback since the user technically clicks on the list item again. If I dont have a way to clear selection like this, then I need to add a button clear the selection which makes the UX clunky.

**Reply by aneeskA** [reply]
My reasoning is that since the List widget already gives selection, I can leverage un-selection as an intuitive UX.

**Reply by andydotxyz** [reply, marked as answer]
It sounds like you are really looking for tapped not selection as a source of information. I would say there are 2 options:

1) Have each item be tappable (button or similar) and handle the indication however you like
2) Extend the List widget and in your tap handling you can see if there is selection and unselect, otherwise pass on the event to the original code.
